### PR TITLE
Feature/edit profile

### LINKED ISSUE #260: Cant edit your family member's profile
Starting point: https://dev.stamacasa.ro/evaluation/other-members/11

**Expected**:

Editing your family member's profile by clicking

> Editare profil

**Actual**:

Click on

> Editare profil

Nothing happens

### PR DESCRIPTION
**Pull request by irinel-nistor**
### What does it fix?

Closes #260

- Updated the api to save and return the PreexistingMedicalCondition  as string array. This was necessary in order to facilitate the population of the fields of the update form
- Added update profile controller
- Enabled edit profile for family member
- Created profile update page

### How has it been tested?

Manually tested the changes 

Issues: Cannot populate county and city, [DropdownSearch ](https://taskforce-fe-components.now.sh/?path=/story/dropdown-search--dropdown-search-closed)component limitation

### CODE REVIEW

File: backend/src/StamAcasa.Common/DTO/UserInfo.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public class UserInfo {
         public string City { get; set; }
         public int Age { get; set; }
         public Gender Gender { get; set; }
-        public string PreexistingMedicalCondition { get; set; }
+        public string[] PreexistingMedicalCondition { get; set; }
```
Comment (mmircea16) [inline backend/src/StamAcasa.Common/DTO/UserInfo.cs:20]: Maybe renaming to `PreexistingMedicalConditions` to show the fact is a list now? Small thing, we can do it later also

**Review by mmircea16 — DISMISSED** [review]
Looks good! 

One question though - what's the plan re: migrations of data in the DB? As it is right now `preexistingMedicalCondition` is a string and, if nothing done, the api call for `/profile` for existing members returns 500. We either nuke the data we have or add a migration script.

**Comment by RaduCStefanescu** [comment]
> Looks good!
> 
> One question though - what's the plan re: migrations of data in the DB? As it is right now `preexistingMedicalCondition` is a string and, if nothing done, the api call for `/profile` for existing members returns 500. We either nuke the data we have or add a migration script.

Is it an issue? As I can see we are changing objects that are used as DTOs, not objects we use for the data model. Can you give us some insight @irinel-nistor ?

**Comment by irinel-nistor** [comment]
> > Looks good!
> > One question though - what's the plan re: migrations of data in the DB? As it is right now `preexistingMedicalCondition` is a string and, if nothing done, the api call for `/profile` for existing members returns 500. We either nuke the data we have or add a migration script.
> 
> Is it an issue? As I can see we are changing objects that are used as DTOs, not objects we use for the data model. Can you give us some insight @irinel-nistor ?

That is correct, we are only changing the objects that are used as DTOs. However I modified the deserialization, so that older versions of the Users.PreexistingMedicalConditions would get deserialized in the same format. This way, we no longer require to add a migration script.

File: frontend/src/components/UpdateProfile/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      />
+    </SidebarLayout>
+  );
+};
+
+UpdateProfile.propTypes = {
+  location: PropTypes.shape({
+    state: PropTypes.shape({
+      id: PropTypes.number,
+      profileDetails: PropTypes.object,
```
Comment (surdu) [inline frontend/src/components/UpdateProfile/index.js]: Shouldn't `id` and `profileDetails` be required ?

Comment (irinel-nistor) [reply frontend/src/components/UpdateProfile/index.js]: Yes, you are right, I'll fix this. Thanks!

**Review by RaduCStefanescu — APPROVED** [review]

**Pull request merged**